cpp/src/arrow/compute/row/compare_internal_avx2.cc

```diff
@@ -236,6 +236,8 @@ uint32_t KeyCompare::CompareBinaryColumnToRowHelper_avx2(
         irow_right =
             _mm256_loadu_si256(reinterpret_cast<const __m256i*>(left_to_right_map) + i);
       }
+      // TODO: Need to test if this gather is OK when irow_right is larger than
```

Comment:
I'll test in the future.

Reply:
When you say "in the future", is it in this PR or another one?

Reply:
Oh sorry, I meant in another PR.